support/builder/build_dep_populate.sh

```diff
@@ -0,0 +1,463 @@
+            credsConfigured=true
+            break
+        fi
+
+        if [[ -n ${AWS_ACCESS_KEY_ID:-} ]]; then
+            if [[ -n ${AWS_SECRET_ACCESS_KEY:-} ]]; then
+                echo ""
+                echo "AWS Credentials configured via ENVVAR detected."
+                echo ""
+                echo "aws_access_key_id=${AWS_ACCESS_KEY_ID:-}"
+                echo "aws_secret_access_key=${AWS_SECRET_ACCESS_KEY:-}"
+                credSelect
+                credsConfigured=true
+                break
+            else
+                echo ""
+                echo "WARNING: Incomplete AWS Credentials configured via ENVVAR."
+                echo "Make sure to set AWS_ACCESS_KEY_ID && AWS_SECRET_ACCESS_KEY"
+                break
+            fi
+        fi
+    done
+
+    if  "${credsConfigured}" ; then
+        echo ""
+        echo "Credentials configured!"
+    else
+        echo ""
+        echo "WARNING: No AWS credentials detected!"
+        read -r -p "Would you like to generate them now? [y/N] " response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            genS3Config
+        else
+            echo "Please configure your AWS credentials and re-run s3migrate."
+            echo ""
+            exit 1
+        fi
+    fi
+}
+
+welcome() {
+    echo ""
+    echo "==========================================================="
+    echo "###########################################################"
+    echo "############## Bldr Build Dep Ingest/Migrate ##############"
+    echo "###########################################################"
+    echo "==========================================================="
+    echo ""
+    echo "This tool will scrape all the hart files from an s3 or minio"
+    echo "bucket and reingest them to populate build dep and tdep metadata"
+    credsConfigured=false
+
+    setBldrUrl
+
+
+
+    if [[ -n "${HAB_AUTH_TOKEN+x}" ]]; then
+        echo ""
+        echo "We were able to discover your builder auth token as: "
+        echo "${HAB_AUTH_TOKEN}"
+        read -r -p "Would you like to use these credentials? [y/N]" response
+            if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                echo "Setting detected credentials"
+            fi
+    else
+        echo ""
+        echo "WARNING: No Builder credentials detected!"
+        read -r -p "Would you like to generate them now? [y/N] " response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            genBldrCreds
+        else
+            echo "Please configure your builder auth-token and re-run s3migrate."
+            echo ""
+            exit 1
+        fi
+    fi
+
+    if [ "$s3type" = "minio" ]; then
+        echo ""
+        echo "It looks like you specified an ingestion from minio!"
+
+        sgroups=("default" "dev" "prod")
+        for i in "${sgroups[@]}"; do
+            if curl -s localhost:9631/services/builder-minio/"${i}" | jq .cfg.key_id > /dev/null; then
```

Comment:
I believe this `if` will evaluate to true so long as the `curl` command doesn't error; regardless of whether `jq .cfg.key_id` matches anything. If you want different behavior consider the `--exit-status` flag to `jq`.

Reply:
What is the point of the `jq .cfg.key_id` here? If won't affect the truth of the conditional.

Also, if we want to avoid curling the same URL three times, we could
```bash
            if minio_output=$(curl -s localhost:9631/services/builder-minio/"${i}"); then
```
then there are two options for populating the vars with the `jq` output:
```bash
                access_key_id=$(echo "$minio_output" | jq .cfg.key_id)
                secret_access_key=$(jq .cfg.secret_key <<< "$minio_output")
```